docs/recipes/react.md

```diff
@@ -280,3 +280,64 @@ class Toggle extends React.Component {
   }
 }
 ```
+
+## Reference hooks safely
+
+When working with React hook we might want to pass a reference to a hook callback that is changing between renders. When done incorrectly you might notice this warning in the console log `Machine given to 'useMachine' has changed between renders.`. This can be addressed in the following ways:
+
+```jsx
+const config = {
+  id: 'machine',
+  /* ... */
+  entry: 'someHookAction', // actions have to be referenced by name
+} as const;
+
+const machine = createMachine(config);
+
+const Component = () => {
+  const doStuff = useSomeHook();
+  const [state, send] = useMachine(machine, React.useMemo(() => {
+    actions: {
+      someHookAction: (ctx) => doStuff(ctx);
+    }
+  }, [doStuff]));
+  return <></>
+}
+```
+
+The above approach won't work if you invoke nested machine. Although a bit more verbose the example below addresses the issue where the config is preserved while the reference to `doStuff` can chance between renders
+
+```jsx
+const configA = {
+  id: 'machineA',
+  /* ... */
+  entry: 'someHookAction', // actions have to be referenced by name
+} as const;
+
+const createMachineA = ({ doStuff }) => createMachine(configA, {
```

Comment:
We don't want to put this into documentation because it makes it harder for our tooling to perform static analysis when visualizing machines, generating typegen information etc.

Reply:
is the tooling using the machine parser? if that is the case from my experience it is not thrown off by a constructor pattern. what would be the alternative. I just wanted to make sure if someone is getting the error in their application a quick google search will lead them to here so they can better understand what the issue is and how it should be addressed